reqs/requirements-dev-lite.txt

```diff
@@ -20,7 +20,7 @@ pytest-timeout>=2.3.1
 pytest-xdist>=3.6.1
 requirements-parser>=0.9.0
 responses>=0.23.1 # requests mocking
-setuptools>=70.0.0 # required for python 3.12
+setuptools>=70.0.0,<71.0.0 # lower bound required for python 3.12, upper bound required for how we use pkg_resources in scripts
```

Comment:
I think this could also go in the constraints-dev file, but I don't want to delay this merge.